pkg/controller/machineset/controller.go

```diff
@@ -150,6 +150,13 @@ func (r *ReconcileMachineSet) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	klog.V(4).Infof("Reconcile machineset %v", machineSet.Name)
+
+	if errList := machineSet.Validate(); len(errList) > 0 {
```

Comment:
we could consider to run and Admission Webhook for this and similar features. This is better than we have today though